packages/@lwc/engine-core/src/framework/vm.ts

```diff
@@ -739,7 +740,11 @@ export function resetComponentRoot(vm: VM) {
     for (let i = 0, len = children.length; i < len; i++) {
         const child = children[i];
 
-        if (!isNull(child) && !isUndefined(child.elm)) {
+        // VFragments are special; their .elm property does not point to the root element since they have no root,
+        // so we have to clean them up differently.
+        if (!isNull(child) && isVFragment(child)) {
```

Comment:
VFragment is the only VNode that represents a set of VNodes and has no root. I couldn't think of a way to deal with the them in a generic way here, so we just check whether this is a Fragment and react accordingly.